src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -234,7 +234,7 @@ void QgsActiveLayerFeaturesLocatorFilter::prepare( const QString &string, const
     }
     else if ( allowNumeric && field.isNumeric() )
     {
-      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue );
+      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue, 0, trunc( numericValue ) == numericValue ? 'f' : 'g' );
```

Comment:
not sure this is the best approach? shall we use `f` in any case?

Reply:
I've no strong opinions either way. Go with your gut 👍